### PR TITLE
[bitnami/nginx] Add `envsubst` to nginx

### LINKED ISSUE #34957: [bitnami/nginx] Add `envsubst` to Nginx Docker image
### Name and Version

bitnami/nginx:1.24

### What is the problem this feature will solve?

Better configurability, better separation of configuration scaffolding and real configuration.

### What is the feature you are proposing to solve the problem?

Th official image uses `envsubst` to be able to pass Env vars into configuration files, see [docs](https://github.com/docker-library/docs/tree/master/nginx#using-environment-variables-in-nginx-configuration-new-in-119).

This allows the inclusion of those files into repositories without exposing deployment specific configuration. This goal isn't reached by the current way of including only files into `/opt/bitnami/nginx/conf/server_blocks/`.

### What alternatives have you considered?

Building a custom image, but it's not an real improvement, compared with the official image.

### PR DESCRIPTION
**Pull request by djschny**
### Description of the change

Addresses #34957

### Benefits

Documented in associated issue.

### Possible drawbacks

Documented in associated issue.

### Applicable issues

Fixes #34957

### CODE REVIEW

**Comment by djschny** [comment]
Switching from draft to regular PR to see if builds will run.

**Comment by javsalgar** [comment]
Hi!

Thank you so much for the PR. The verification is not being triggered because of this

```
Changes should affect to only one asset. You are currently modifying: kubectl nginx
```

Could you split the PR for `kubectl` and `nginx`?

**Comment by djschny** [comment]
> Could you split the PR for kubectl and nginx?

Sure, sorry didn't realize that.

**Comment by djschny** [comment]
@javsalgar all fixed, should be good to go now on this one, will open a second PR.

**Comment by gongomgra** [comment]
Hi @djschny,

As mentioned in #55365 , we have created an internal task to evaluate the `gettext-base` package addition in all distributions in which we provide NGINX images as part of [Tanzu Application Catalog](https://tanzu.vmware.com/application-catalog). We will keep you posted.

In the meantime, please extend our image with the required packages on your side.

**Comment by djschny** [comment]
Requested changes done.

**Comment by djschny** [comment]
@gongomgra is there anything else I need to change to get this moving?

**Comment by gongomgra** [comment]
Hi @djschny,

Sorry for the delay. I have been checking this and I have found we are adding the `gettext` package instead of `gettext-base` in other images like [appsmith](https://github.com/bitnami/containers/blob/main/bitnami/appsmith/1/debian-12/Dockerfile#L26). Can you please update your PR? I don't have any other suggestion/requirement to this.

File: bitnami/nginx/1.24/debian-11/Dockerfile [hunk 1]
```diff
@@ -23,7 +23,7 @@ ENV HOME="/" \
 COPY prebuildfs /
 SHELL ["/bin/bash", "-o", "errexit", "-o", "nounset", "-o", "pipefail", "-c"]
 # Install required system packages and dependencies
-RUN install_packages ca-certificates curl libcrypt1 libgeoip1 libpcre3 libssl1.1 openssl procps zlib1g
+RUN install_packages ca-certificates curl libcrypt1 libgeoip1 libpcre3 libssl1.1 openssl procps zlib1g gettext-base
```
Comment (gongomgra) [inline bitnami/nginx/1.24/debian-11/Dockerfile:26]: Please use `gettext` as in other images like [appsmith](https://github.com/bitnami/containers/blob/main/bitnami/appsmith/1/debian-12/Dockerfile#L26)

**Review by gongomgra — CHANGES_REQUESTED** [review]
Please also fix the conflicts with main branch

File: bitnami/nginx/1.25/debian-11/Dockerfile [hunk 2]
```diff
@@ -23,7 +23,7 @@ ENV HOME="/" \
 COPY prebuildfs /
 SHELL ["/bin/bash", "-o", "errexit", "-o", "nounset", "-o", "pipefail", "-c"]
 # Install required system packages and dependencies
-RUN install_packages ca-certificates curl libcrypt1 libgeoip1 libpcre3 libssl1.1 openssl procps zlib1g
+RUN install_packages ca-certificates curl libcrypt1 libgeoip1 libpcre3 libssl1.1 openssl procps zlib1g gettext-base
```
Comment (jotamartos) [inline bitnami/nginx/1.25/debian-11/Dockerfile:26]:
```suggestion
RUN install_packages ca-certificates curl gettext libcrypt1 libgeoip1 libpcre3 libssl1.1 openssl procps zlib1g
```

**Comment by gongomgra** [comment]
Sorry @djschny, I have just noticed your changes are for an old version of the Dockerfiles based on debian-11, but we have recently updated the base image to debian-12. I have added the `gettext` package for the next image revision, so I'm closing this PR. Sorry for the inconveniences.

**Pull request closed without merging**